app/main.py

```diff
@@ -1,2 +1,108 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._length: int = 0
+        self._hash_table: list = [None] * 8
+        self._hash_size: int = 8
+        self._resize_factor: float | int = 2 / 3
+        self._resize_multiplier: int = 2
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self._hash_size_check_increase()
+        key_hash_position = self._find_key_position(key)
+
+        while self._hash_table[key_hash_position]:
+            if self._hash_table[key_hash_position][0] == key:
+                self._hash_table[key_hash_position][1] = value
+                return print(f"value with key '{key}' were rewritten")
+            key_hash_position = (key_hash_position + 1) % self._hash_size
```

Comment:
You repeat this block of code on line 31, 50 so move it to a function to make your code DRY